nano/node/blockprocessor.hpp

```diff
@@ -103,21 +122,21 @@ class block_processor final
 	void queue_unchecked (store::write_transaction const &, nano::hash_or_account const &);
 	processed_batch_t process_batch (nano::unique_lock<nano::mutex> &);
 	context next ();
-	void add_impl (context);
+	bool add_impl (context, std::shared_ptr<nano::transport::channel> const & channel = nullptr);
```

Comment:
Is it worth putting channel inside context?

Reply:
I don't know, maybe? If we were to implement peer scoring it could be useful, though I'm not sure if we should do that now.

Reply:
Knowing the channel that a block arrived from sounds like a good thing to keep around.
However, we might run into object lifetime problems... I am not sure either.
We could leave it until we need it.